cost-analyzer/templates/aggregator-cloud-cost-deployment-template.yaml

```diff
@@ -0,0 +1,140 @@
+{{- if .Values.kubecostAggregator.cloudCost.enabled }}
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "cloudCost.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{ include "cloudCost.commonLabels" . | nindent 4 }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      {{ include "cloudCost.selectorLabels" . | nindent 6 }}
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: cloud-cost
+        app.kubernetes.io/instance: {{ .Release.Name }}
+        app: cloud-cost
+    spec:
+      restartPolicy: Always
+      serviceAccountName: {{ template "cloudCost.serviceAccountName" . }}
+      volumes:
+        {{- if .Values.kubecostModel.etlBucketConfigSecret }}
+        - name: etl-bucket-config
+          secret:
+            defaultMode: 420
+            secretName: {{ .Values.kubecostModel.etlBucketConfigSecret }}
+        {{- end }}
+        {{- if .Values.kubecostModel.federatedStorageConfigSecret }}
+        - name: federated-storage-config
+          secret:
+            defaultMode: 420
+            secretName: {{ .Values.kubecostModel.federatedStorageConfigSecret }}
+        {{- end }}
+        {{- if .Values.kubecostProductConfigs.cloudIntegrationSecret }}
+        - name: cloud-integration
+          secret:
+            secretName: {{ .Values.kubecostProductConfigs.cloudIntegrationSecret }}
+            items:
+              - key: cloud-integration.json
+                path: cloud-integration.json
+        {{- else }}
+        {{- fail "Cloud Cost requires configuration secret" }}
+        {{- end }}
+      containers:
+        - name: cloud-cost
+          {{- if .Values.kubecostModel }}
+          {{- if .Values.kubecostModel.openSourceOnly }}
+          {{- fail "Kubecost Aggregator cannot be used with open source only" }}
+          {{- else if .Values.kubecostAggregator.fullImageName }}
+          image: {{ .Values.kubecostAggregator.fullImageName }}
+          {{- else if .Values.kubecostModel.fullImageName }}
+          image: {{ .Values.kubecostModel.fullImageName }}
+          {{- else if .Values.imageVersion }}
+          image: {{ .Values.kubecostModel.image }}:{{ .Values.imageVersion }}
+          {{- else }}
+          image: {{ .Values.kubecostModel.image }}:prod-{{ $.Chart.AppVersion }}
+          {{ end }}
+          {{- else }}
+          image: gcr.io/kubecost1/cost-model:prod-{{ $.Chart.AppVersion }}
+          {{ end }}
+          readinessProbe:
+            httpGet:
+              path: /healthz
+              port: 9005
+            initialDelaySeconds: 10
+            periodSeconds: 5
+            failureThreshold: 200
+          imagePullPolicy: Always
+          args: ["cloud-cost"]
+          ports:
+            - name: tcp-api
+              containerPort: 9005
+              protocol: TCP
+          resources:
+{{ toYaml .Values.kubecostAggregator.cloudCost.resources | indent 12 }}
```

Comment:
Let's make this indentation easier, please. Templating should be consistent with the indentation of the field where it applies.

Reply:
fixed